src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/SamServiceAccountProvider.scala

```diff
@@ -17,7 +20,12 @@ abstract class SamServiceAccountProvider(config: Config) extends ServiceAccountP
   implicit val materializer = ActorMaterializer()
   implicit val executionContext = system.dispatcher
 
+
   protected lazy val samConfig = config.as[SamConfig]("sam")
+  protected lazy val cacheExpiryTime = toScalaDuration(config.getConfig("sam").getDuration("cacheExpiryTime"))
```

Comment:
Is there a better way to get these configs?

Reply:
You can do `config.getConfig("some.dotted.path")` to yank inner objects from a config block. But it looks like what you're doing here doesn't match the layout of the configs. The value `config` passed in the class constructor is whatever `auth.providerConfig` points to in `leonardo.conf`. Looking at the test version of `reference.conf` that doesn't match what you're unpacking here.